packages/wonder-blocks-core/src/hooks/use-online.js

```diff
@@ -0,0 +1,29 @@
+// @flow
+import {useForceUpdate} from "@khanacademy/wonder-blocks-core";
+import {useEffect} from "react";
+
+/**
+ * Track the online status of the browser.
+ *
+ * This hook monitors the offline and online events, forcing the consuming
+ * component to re-render when they fire.
+ *
+ * @returns {boolean} The current value of `navigator.onLine`.
+ */
+export const useOnline = (): boolean => {
+    const forceUpdate = useForceUpdate();
+
+    useEffect(() => {
+        const handleChange = () => forceUpdate();
+
+        window.addEventListener("online", handleChange);
+        window.addEventListener("offline", handleChange);
+
+        return () => {
+            window.removeEventListener("online", handleChange);
+            window.removeEventListener("offline", handleChange);
+        };
+    }, [forceUpdate]);
+
+    return navigator.onLine;
```

Comment:
**explanation:** Seems silly to store the value in state when it's already got a source of truth. So instead, we use force updates on the events but always use `navigator.onLine` as the actual value.